src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +59,94 @@
     }
   }
 
+  /**
+   * Get the historical price data for the passed in symbol over the passed in range.
+   *
+   * @param symbol stock symbols to get historical data for.
+   * @param range selection of time for which to get historical data.
+   * @return a list of historical price objects for the Symbol that is passed in covering the date
+   *     range.
+   */
+  public List<IexHistoricalPrice> getHistoricalPricesForSymbol(final String symbol,
```

Comment:
This method is very long, please break it down into smaller components. It's difficult to follow the logic in this form.